homework-submission/react-hyf/react-hyf-client/src/Components/fetchdata.js

```diff
@@ -0,0 +1,38 @@
+import React, { useState, useEffect } from 'react';
+
+function DataFetching(source) {
+  const [loading, setLoading] = useState(true);
+  const [randomNumber, setRandomNumber] = useState(0);
+  const [results, setResults] = useState([]);
+  const [error, setError] = useState('');
+
+  useEffect(() => {
+    async function fetchData() {
+      try {
+        const data = await fetch(source);
+        const json = await data.json();
+
+        if (json) {
+          setLoading(false);
+          setResults(json);
+        }
+      } catch (error) {
+        setLoading(false);
+        setError(error.message);
+      }
+
+      setLoading(false);
+    }
+
+    fetchData();
+  }, [source, randomNumber]);
+
+  return {
+    error,
+    loading,
+    results,
+    setRandomNumber,

```

Comment:
`setRandomNumber` feels like a strange method name to trigger fetching new data. A nice addition would be that you return a function here called `trigger`, which automatically generates a random number and feeds that into `setRandomNumber`.

Reply:
also fixed this :)